pkg/taskservice/task_service_holder.go

```diff
@@ -305,6 +305,9 @@ func (s *refreshableTaskStorage) refreshTask(ctx context.Context) {
 			return
 		case lastAddress := <-s.refreshC:
 			s.mu.Lock()
+			if s.mu.closed {
```

Comment:
wait, don't you need to defer s.mu.Unlock?

Reply:
fixed.